### PR TITLE
Rewrite the README so it is closer to the gax-java one

### PR DESCRIPTION
**Pull request by jskeet**
This also adds a build status badge, in anticipation of Travis being enabled on the repo very soon :)
I've preserved the details of what's going to be in which package, with the hope that we'll move it
to wider documentation at some point soon.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,56 @@
...
 aids the development of APIs, client and server, based on
-[GRPC](http://grpc.io) and Google API conventions.
+[gRPC](http://grpc.io) and Google API conventions.
 
 Application code will rarely need to use most of the classes within this
-library directly, but code generated automatically from the API definition
-files can use services such as page streaming and request bundling to provide
+library directly, but code generated automatically from the [API definition
+files](https://github.com/googleapis/googleapis/)
+can use services such as page streaming and request bundling to provide
```
Comment (chrisdunelm) [inline README.md]: Or possibly not request bundling

File: README.md [hunk 2]
```diff
@@ -1,13 +1,56 @@
...
 a more convenient and idiomatic API surface to callers.
 
+Quickstart
+---
+
+Currently there is no public package for this library. When we come
+closer to release, it will be published on
+[NuGet](https://nuget.org) with an expected package name of
+`Google.Api.Gax`. However, it will also be dependency of specific
```
Comment (chrisdunelm) [inline README.md]: ...it will also be *a* dependency of...

Comment (jskeet) [inline README.md]: Fixed, thanks. (Will squash, push and merge when I've fixed any other comments.)

**Comment by chrisdunelm** [comment]
LGTM

File: README.md (hunk 1 above)
Comment (jskeet) [inline README.md]: True. Will stick to just page streaming for the moment.

**Pull request merged**